contracts/masset/liquidator/Liquidator.sol

```diff
@@ -0,0 +1,263 @@
+        liq.bAsset = _bAsset;
+        liq.integration = _integration;
+        liq.sellToken = _sellToken;
+        liq.uniswapPath = _uniswapPath;
+        liq.amount = _amount;
+        liq.paused = _paused;
+
+        emit LiquidationPaused(_bAsset);
+    }
+
+    /**
+    * @dev Delete a liquidation
+    * @param _bAsset The _bAsset for the liquidation
+    */
+    function deleteLiquidation(address _bAsset)
+        external
+        onlyGovernor
+    {
+        require(liquidations[_bAsset].bAsset != address(0), "No liquidation for this bAsset");
+
+        delete liquidations[_bAsset];
+        emit LiquidationRemoved(_bAsset);
+    }
+
+    /**
+    * @dev  Update the Uniswap contract address
+    *       Whilst it is unlikely this will be needed it is helpful for testing
+    * @param _uniswapAddress The uniswap address to assign
+    */
+    function updateUniswapAddress(address _uniswapAddress)
+        external
+        onlyGovernor
+    {
+
+        require(_uniswapAddress != address(0), "_uniswapAddress cannot be zero address");
+        uniswapAddress = _uniswapAddress;
+        emit UniswapUpdated(_uniswapAddress);
+    }
+
+    /**
+    * @dev Allows a calling integration contract to collect redeemed tokens
+    */
+    function collect(address _bAsset)
+        external   
+    {
+        Liquidation memory liq = liquidations[_bAsset];
+        require(msg.sender == liq.integration, "Only integration contract can execute");
+
+        address sendAddress = liq.integration;
+        address asset = liq.sellToken;
+        
+        uint256 balance = IERC20(asset).balanceOf(address(this));
+        IERC20(asset).safeApprove(sendAddress, 0);
+        IERC20(asset).safeApprove(sendAddress, balance);
+        IERC20(asset).safeTransfer(sendAddress, balance);
+    }
+
+    /**
+    * @dev Triggers a liquidation
+    * @param _bAsset The _bAsset liquidation to be triggered
+    */
+    function triggerLiquidation(address _bAsset)
+        external
+    {
+        Liquidation memory liquidation = liquidations[_bAsset];
+
+        require(liquidation.lastCalled + 1 days > now, "Trigger liquidation only callable every 24 hours");
+
+        address asset = liquidation.sellToken;
+        uint256 balance = IERC20(asset).balanceOf(address(this));
+        require(balance != 0, "No tokens to liquidate");
+
+        uint256 sellAmount;
+
+        uint256 randomAmountToSell = uint(blockhash(block.number-1)) % 3000 + 1000;
+
+        if (balance > sellAmount) {
+            sellAmount = randomAmountToSell;
+        } else {
+            sellAmount = balance;
+        }
+
+        IERC20(asset).safeApprove(uniswapAddress, 0);
+        IERC20(asset).safeApprove(uniswapAddress, sellAmount);
+
+        IUniswapV2Router02(uniswapAddress).swapExactTokensForTokens(
+            sellAmount,     
+            uint256(0),
+            liquidation.uniswapPath,
+            address(this),
+            now.add(1800)
+        );
+
```

Comment:
We need to deposit the newly purchased bAsset into the corresponding lending market here, so that the pToken is ready for collection